code/game/objects/items/jetpack.dm

```diff
@@ -0,0 +1,174 @@
+#define FUEL_USE 5
+#define JETPACK_COOLDOWN_TIME 10 SECONDS
+
+/obj/item/jetpack_marine
+	name = "marine jetpack"
+	desc = "A high powered jetpack with enough fuel to send a person flying for a short while. It allows for fast and agile movement on the battlefield. <b>Alt right click to fly to a destination when the jetpack is equipped.</b>"
+	icon = 'icons/obj/items/jetpack.dmi'
+	icon_state = "jetpack_marine"
+	w_class = WEIGHT_CLASS_BULKY
+	flags_equip_slot = ITEM_SLOT_BACK
+	///maximum amount of fuel in the jetpack
+	var/fuel_max = 60
+	///current amount of fuel in the jetpack
+	var/fuel_left = 60
+	///threshold to change the jetpack fuel indicator
+	var/fuel_indicator = 40
+	///How quick you will fly (warning, it rounds up to the nearest integer)
+	var/speed = 1
+	///How long the jetpack allows you to fly over things
+	var/hovering_time = 1 SECONDS
+	///True when jetpack has flame overlay
+	var/lit
+	///True if you can use shift click/middle click to use it
+	var/selected
+
+/obj/item/jetpack_marine/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/item/jetpack_marine/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	if(!ishuman(user))
+		return
+	if(fuel_left == 0)
+		to_chat(user, "The fuel gauge beeps out, it has no fuel left")
+	else
+		to_chat(user, "The fuel gauge meter indicate it has [fuel_left/FUEL_USE] uses left")
+	icon = 'icons/obj/items/jetpack.dmi'
+	icon_state = "jetpack_marine"
+	update_icon() //The animation breaks often in high pop, this tries to fix it.
+	user.update_inv_back()//Not sure it'll work, i don't understand what is causing the bug anyway
+
+/obj/item/jetpack_marine/equipped(mob/user, slot)
+	. = ..()
+	if(slot == SLOT_BACK)
+		RegisterSignal(user, COMSIG_MOB_CLICK_ALT_RIGHT, .proc/try_to_use_jetpack)
+		var/datum/action/item_action/toggle/action = new(src)
+		action.give_action(user)
+
+/obj/item/jetpack_marine/dropped(mob/user)
+	. = ..()
+	UnregisterSignal(user, list(COMSIG_MOB_CLICK_ALT_RIGHT, COMSIG_MOB_MIDDLE_CLICK))
+	UnregisterSignal(user, COMSIG_ITEM_EXCLUSIVE_TOGGLE)
+	selected = FALSE
+	actions.Cut()
+
+/obj/item/jetpack_marine/ui_action_click(mob/user, datum/action/item_action/action)
```

Comment:
document new procs

Reply:
This ain't a new proc chief, the new undocumented proc was 10 lines after :)